### ISSUE TITLE
router.back() fails when using experimental basePath feature

### ISSUE THREAD

**Issue by cederigo** [zeit/next.js#11365]
# Bug report

Navigation breaks when using the experimental basePath feature in combination with dynamic routes and `router.back()`

I had a look at: https://github.com/zeit/next.js/blob/b274fe39d38111588e625342a89dc568ad38fff5/packages/next/next-server/lib/router/router.ts#L419

And I think `parse(as)` should be replace with `parse(addBasePath(as))` but I not so sure.

## Steps to reproduce

1. Use next.config.js as below
2. BASE_PATH=/test yarn dev
3. router.push('/articles/[id]', '/articles/1') // At this point router.back() works
4. router.push('/articles/[id]/edit', '/articles/1/edit') // router.back() throws error

## Expected behavior

`router.back()` navigates back as expected.

## Screenshots

<img width="781" alt="Screenshot 2020-03-26 at 11 01 41" src="https://user-images.githubusercontent.com/415316/77635116-5ad2fe80-6f52-11ea-84e3-3b36d25002d7.png">

## System information

- OS: macOS & Windows
- Browser (if applies) [chrome & edge]
- Version of Next.js: 9.2.1

## Additional context

next.config.js:
```
const BASE_PATH = process.env.BASE_PATH || ''
module.exports = {
  assetPrefix: BASE_PATH,
  experimental: {
    basePath: BASE_PATH
  }
}
```

**Comment by Timer** [comment]
This should be fixed as of 9.3.5.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
